### PR TITLE
ALP-140 added handler for random spaces

### PR DESCRIPTION
**Pull request by RamsesDijkstra**
Added a separate field that holds the array of random spaces, with a handler that selects one of them at random

### CODE REVIEW

File: universe/common/helper/worlds.go [hunk 1]
```diff
@@ -72,6 +76,21 @@ func AddWorldFromTemplate(worldTemplate *WorldTemplate, updateDB bool) (uuid.UUI
 		}
 	}
 
+	// adding random space
+	rand.Seed(time.Now().Unix())
```
Comment (cnaize) [inline universe/common/helper/worlds.go]: it's not a cheap operation
you need to check is it really necessary, if yes, then move it to `main.go`

File: universe/common/helper/worlds.go [hunk 2]
```diff
@@ -72,6 +76,21 @@ func AddWorldFromTemplate(worldTemplate *WorldTemplate, updateDB bool) (uuid.UUI
...
+	randomSpace := worldTemplate.RandomSpaces[rand.Intn(len(worldTemplate.RandomSpaces))]
```
Comment (cnaize) [inline universe/common/helper/worlds.go]: I think it's better to just add your random space to `worldTemplate.Spaces` and let it to be added as regular child

File: universe/common/helper/spaces.go [hunk 3]
```diff
@@ -27,6 +27,7 @@ type SpaceTemplate struct {
 	Label           *string              `json:"label"`
 	SpaceAttributes []*entry.Attribute   `json:"space_attributes"`
 	Spaces          []*SpaceTemplate     `json:"spaces"`
+	RandomSpaces    []*SpaceTemplate     `json:"random_spaces"`
```
Comment (cnaize) [inline universe/common/helper/spaces.go:30]: are you sure that it supposed to be part of `SpaceTemplate` not `WorldTemplate`?

**Review by cnaize — CHANGES_REQUESTED** [review]

File: universe/common/helper/spaces.go (hunk 3 above)
Comment (RamsesDijkstra) [reply universe/common/helper/spaces.go:30]: yes, they are supposed to be children of the world

**Review by terbooter — APPROVED** [review]

**Pull request merged**